### PR TITLE
9016 - Updated dissolution filer

### PR DESCRIPTION
**Pull request by andrepestana-aot**
*Issue #:* /bcgov/entity#9016
*Description of changes:*

- updated dissolution filer processor; 
- added tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by argush3 — CHANGES_REQUESTED** [review]
I think we need to add a processor for special resolutions for COOP voluntary dissolutions.  I'm thinking it might only require adding meta data to the filing.

File: queue_services/entity-filer/src/entity_filer/filing_processors/dissolution.py [hunk 1]
```diff
@@ -18,24 +18,28 @@
...
 
     logger.debug('processing dissolution: %s', filing)
     dissolution_date = datetime.fromisoformat(dissolution_filing.get('dissolutionDate'))
-    # Currently we don't use this for anything?
+    # FUTURE: remove hasLiabilities from schema as it can be derived from dissolutionStatementType
     # has_liabilities = filing['dissolution'].get('hasLiabilities')
+
     business.dissolution_date = dissolution_date
 
     # remove all directors and add custodial party if in filing
```
Comment (thorwolpert) [inline queue_services/entity-filer/src/entity_filer/filing_processors/dissolution.py:45]: Could you do me favour and change this comment, directors are not removed and the custodian is added.
It's different than the other party updates.

File: queue_services/entity-filer/src/entity_filer/filing_processors/dissolution.py [hunk 2]
```diff
@@ -57,6 +61,27 @@ def process(business: Business, filing: Dict, filing_meta: FilingMeta):
         court_order_json = dpath.util.get(dissolution_filing, '/courtOrder')
         filings.update_filing_court_order(filing, court_order_json)
 
+    if filing['business']['legalType'] == Business.LegalTypes.COOP.value:
```
Comment (thorwolpert) [inline queue_services/entity-filer/src/entity_filer/filing_processors/dissolution.py:64]: you don't need to add the ```.value```

**Review by thorwolpert — COMMENTED** [review]
couple of small changes

**Comment by thorwolpert** [comment]
Re: meta additions, that would be good to place the sub-type of the dissolution there for the ledger and other views.

**Comment by andrepestana-aot** [comment]
Sorry, I messed up the branch. I created a fresh one: https://github.com/bcgov/lear/pull/1343

**Pull request closed without merging**